packages/playground/src/data-boundary.tsx

```diff
@@ -9,8 +9,8 @@ export function DataBoundary({
   dataSource,
   path,
   children,
-}: PropsWithChildren<{ dataSource: WebSocketDataSource; path: string }>) {
-  const state = useStream(dataSource.state)
+}: PropsWithChildren<{ dataSource: WebSocketDataSource | DataSource; path: string }>) {
+  const state = useStream('state' in dataSource ? dataSource.state : undefined)
```

Comment:
fix: typescript error when passing HTTPDataSource to Boundary.

The extra conditional is just to make TypeScript happy. `useStream` handles undefined already. That's why previous code was working just fine.

Unfortunately, `dataSource?.state` is going to fail too.